examples/devices/lighty-network-topology-device/src/main/java/io/lighty/netconf/device/topology/rpcs/NetworkTopologyServiceImpl.java

```diff
@@ -456,52 +460,55 @@ public void setDataBrokerService(final DataBroker dataBrokerService) {
         this.executor.submit(new Callable<RpcResult<GetTopologiesOutput>>() {
             @Override
             public RpcResult<GetTopologiesOutput> call() throws Exception {
-                final ReadTransaction readTx =
-                    NetworkTopologyServiceImpl.this.dataBrokerService.newReadOnlyTransaction();
-                final InstanceIdentifier<NetworkTopology> tii =
-                        InstanceIdentifier.builder(NetworkTopology.class).build();
-                final Optional<NetworkTopology> networkTopology = readTx.read(LogicalDatastoreType.CONFIGURATION, tii)
-                        .get(TimeoutUtil.TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
-                if (networkTopology.isPresent()) {
-                    final org.opendaylight.yang.gen.v1.urn.tech.pantheon.netconfdevice.network.topology.rpcs
-                        .rev180320.topology.data.TopologyBuilder topologyBuilder =
-                            new org.opendaylight.yang.gen.v1.urn.tech.pantheon.netconfdevice.network.topology.rpcs
-                                .rev180320.topology.data.TopologyBuilder();
-                    final Collection<Topology> topologyList = networkTopology.get().nonnullTopology().values();
-                    final List<org.opendaylight.yang.gen.v1.urn.tech.pantheon.netconfdevice.network.topology.rpcs
-                        .rev180320.topology.data.Topology> topologyListFinal =
-                            new ArrayList<>();
-                    for (final Topology t : topologyList) {
-                        final List<Node> nodeList = new ArrayList<>();
-                        if (t.getNode() != null) {
-                            for (final org.opendaylight.yang.gen.v1.urn.tbd.params.xml.ns.yang.network.topology
-                                .rev131021.network.topology.topology.Node n : t.nonnullNode().values()) {
-                                final NodeKey nk = new NodeKey(n.getNodeId());
-                                final Node nb = buildNode(nk, n);
-                                nodeList.add(nb);
+                try (ReadTransaction readTx =
+                             NetworkTopologyServiceImpl.this.dataBrokerService.newReadOnlyTransaction()) {
+                    final InstanceIdentifier<NetworkTopology> tii =
+                            InstanceIdentifier.builder(NetworkTopology.class).build();
+                    final Optional<NetworkTopology> networkTopology =
+                            readTx.read(LogicalDatastoreType.CONFIGURATION, tii)
```

Comment:
Why resource-try is not close after resource is used? Fix other resource-try to not cover whole method but only required things.